tests/test_init.py

```
@@ -5580,40 +5421,32 @@ async def test_reconciliation_cleanup_unload_outcomes(
-async def test_unload_setup_platforms_after_reconciliation_failure_returns_false_on_unload_exception(
+@pytest.mark.parametrize(
+    "unload_result",
+    [
+        pytest.param(False, id="false-result"),
+        pytest.param(HomeAssistantError("unload error"), id="handled-error"),
+    ],
+)
+async def test_unload_setup_platforms_after_reconciliation_failure_returns_false_for_failures(
     ph_hass: Any,
     make_config_entry: Callable[..., MockConfigEntry],
+    unload_result: bool | BaseException,
```

Ruff (0.16.1)

[warning] Boolean-typed positional argument in function definition

(FBT001)